compair/static/modules/assignment/assignment-module.js

```diff
@@ -218,6 +278,101 @@ module.factory(
     }
 ]);
 
+module.factory(
+    "actionElementPermissions", function (){
```

Comment:
I'm hoping that we can create one of these for `course`, `assignment`, and `answer`.

It should be a bit more generic so that you can pass in assignment + assignment status and get a boolean response to different kinds of questions/states.

ex: is the current user in the answer period for the given assignment?
```
isAnswerPeriod: function(assignment, assignmentStatus, [maybe canManageAssignment]) {
    var answerPeriod = false;

    //check conditions
    ...

   return answerPeriod; //or return a promise if we don't pass permissions along
}
```

Reply:
We already have things like `assignment.answer_period` and `assignment.compare_period` that answer those types of questions. I don't think it's necessary to create functions for those types of things. If you look at the overall strategy I have behind this, you can see that I'm trying to minimise conditionals in the view, and instead define them cleanly in this directive.

Reply:
bad example on my end but the general idea holds, right?

Reply:
Do we need to be able to ask about the state? I don't see how we would benefit from that. We have a lot of different states, so that's why I think it makes more sense to rather just pass in the element you want to view, and just ask if it should be displayed or not. That way, we don't have to define conditionals in different parts of the application, rather we will have all of them in one central place.